examples/node-regression-with-subgraph-and-graph-sample.ipynb

```diff
@@ -0,0 +1,713 @@
+      "File \u001B[0;32m/opt/homebrew/Caskroom/miniconda/base/lib/python3.9/site-packages/neo4j/io/_common.py:244\u001B[0m, in \u001B[0;36mResponse.on_failure\u001B[0;34m(self, metadata)\u001B[0m\n\u001B[1;32m    242\u001B[0m \u001B[38;5;28;01mif\u001B[39;00m callable(handler):\n\u001B[1;32m    243\u001B[0m     handler()\n\u001B[0;32m--> 244\u001B[0m \u001B[38;5;28;01mraise\u001B[39;00m Neo4jError\u001B[38;5;241m.\u001B[39mhydrate(\u001B[38;5;241m*\u001B[39m\u001B[38;5;241m*\u001B[39mmetadata)\n",
+      "\u001B[0;31mClientError\u001B[0m: {code: Neo.ClientError.Procedure.ProcedureCallFailed} {message: Failed to invoke procedure `gds.graph.project`: Caused by: java.lang.IllegalArgumentException: A graph with name 'wiki_animals' already exists.}"
+     ]
+    }
+   ],
+   "source": [
+    "# First, let's project our graph into the GDS Graph Catalog\n",
+    "# We will use a native projection to begin with\n",
+    "G_animals, projection_result = gds.graph.project(\n",
+    "    \"wiki_animals\",\n",
+    "    [\"Chameleon\", \"Squirrel\", \"Crocodile\"],\n",
+    "    {\"LINK\": {\"orientation\": \"UNDIRECTED\"}},\n",
+    "    nodeProperties=[\"features_one_hot\", \"target\"],\n",
+    ")\n",
+    "print(projection_result[[\"graphName\", \"nodeCount\", \"relationshipCount\"]])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 506,
+   "id": "e4c1526a",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "computeMillis     47\n",
+      "componentCount     3\n",
+      "Name: 0, dtype: object\n"
+     ]
+    }
+   ],
+   "source": [
+    "# A good first analysis is to inspect connectivity\n",
+    "# We use the WCC algorithm to see how many components we have\n",
+    "wcc_result = gds.wcc.mutate(G_animals, mutateProperty=\"wcc_component\")\n",
+    "\n",
+    "print(wcc_result[[\"computeMillis\", \"componentCount\"]])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 507,
+   "id": "cbd63cd3",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "fromGraphName                wiki_animals\n",
+      "nodeFilter            n.wcc_component = 0\n",
+      "relationshipFilter                      *\n",
+      "graphName             chameleon_component\n",
+      "nodeCount                            2277\n",
+      "relationshipCount                   72202\n",
+      "projectMillis                          88\n",
+      "Name: 0, dtype: object\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Now, we can use the subgraph projection to separate out only one of the components\n",
+    "G_chameleon, subgraph_result = gds.beta.graph.project.subgraph(\n",
+    "    \"chameleon_component\",\n",
+    "    G_animals,\n",
+    "    \"n.wcc_component = 0\",  # Component with id 0 corresponds to the Chameleon part of the graph\n",
```

Comment:
how do they know that ? perhaps write a cyperquery that shows the max and min component number of the chameleon nodes to be `0` ?

Reply:
Due to tie-breaking rules, there will always be a component with id 0 (lower id wins). But we don't know what labels that indicate without Cyphering. I will add something.

Reply:
Ach, I don't want to use Cypher in the notebook tho -- I said above it was pure Python.

Reply:
OK I ended up expanding this part quite a bit.